compiler/quilt/tools/store.py

```diff
@@ -230,6 +252,29 @@ def cache_path(self, name):
         """
         return os.path.join(self._path, self.CACHE_DIR, name)
 
+    def prune(self):
+        """
+        Clean up all objects not referenced by any packages.
+        """
+        objdir = os.path.join(self._path, self.OBJ_DIR)
+        all_obj = os.listdir(objdir)
+
+        for pkg in self.iterpackages():
+            for node in pkg.get_contents().preorder():
+                # TODO: the or below isn't scalable. Add a common baseclass for
+                # File and Table nodes like DataNode in nodes.py.
+                if isinstance(node, (FileNode, TableNode)):
+                    for objhash in node.hashes:
+                        all_obj.remove(objhash)
+
+        for obj in all_obj:
+            print("Removing: {0}".format(obj))
+            os.remove(self.object_path(obj))
```

Comment:
What happens to the build cache?

Reply:
Good catch. The safest thing seems like checking that objects exist before using a cache entry. I added that and a unit test to catch this bug.